session/adodb-session2.php

```diff
@@ -564,29 +564,44 @@ static function open($save_path, $session_name, $persist = null)
 #		assert('$database');
 #		assert('$driver');
 #		assert('$host');
-
-		$conn = ADONewConnection($driver);
-
-		if ($debug) {
-			$conn->debug = true;
-			ADOConnection::outp( " driver=$driver user=$user db=$database ");
-		}
-
-		if (empty($conn->_connectionID)) { // not dsn
+		if (strpos($driver, 'pdo_') === 0){
+			$conn = ADONewConnection('pdo');
+			$driver = str_replace('pdo_', '', $driver);
+			$dsn = $driver.':'.'hostname='.$host.';dbname='.$database.';';
 			if ($persist) {
 				switch($persist) {
 				default:
-				case 'P': $ok = $conn->PConnect($host, $user, $password, $database); break;
-				case 'C': $ok = $conn->Connect($host, $user, $password, $database); break;
-				case 'N': $ok = $conn->NConnect($host, $user, $password, $database); break;
+				case 'P': $ok = $conn->PConnect($dsn,$user,$password); break;
+				case 'C': $ok = $conn->Connect($dsn,$user,$password); break;
+				case 'N': $ok = $conn->NConnect($dsn,$user,$password); break;
 				}
 			} else {
-				$ok = $conn->Connect($host, $user, $password, $database);
+				$ok = $conn->Connect($dsn,$user,$password);
+			}
+		}else{
+			$conn = ADONewConnection($driverÏ);
```

Comment:
discussion #395 I think here is the bad char, the I with 2 dots after $driver.

Reply:
That odd letter 'I' is not in the version I have, it should not be there.